src/audio/module_adapter/module_adapter.c

```diff
@@ -241,20 +241,33 @@ int module_adapter_prepare(struct comp_dev *dev)
 	}
 
 	/* allocate memory for input buffers */
-	mod->input_buffers = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM,
-				     sizeof(*mod->input_buffers) * mod->num_input_buffers);
-	if (!mod->input_buffers) {
-		comp_err(dev, "module_adapter_prepare(): failed to allocate input buffers");
-		return -ENOMEM;
+	if (!mod->num_input_buffers) {
+		mod->input_buffers = NULL;
+	} else {
+		mod->input_buffers =
+			rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM,
+				sizeof(*mod->input_buffers) * mod->num_input_buffers);
+
+		if (!mod->input_buffers) {
+			comp_err(dev, "module_adapter_prepare(): failed to allocate input buffers");
+			ret = -ENOMEM;
+			goto in_free;
```

Comment:
Hmm, "in_free" will rfree input_buffers, but that alloc already failed, so this does not make sense. 
Instead "return -ENOMEM;" directly here?

Reply:
Fixed